src/lib/services/proxy-service.ts

```diff
@@ -188,7 +189,7 @@ export class ProxyService {
         } catch (error) {
             this.logger.debug('Unable to fetch frontend settings', error);
         }
-        return this.cachedFrontendSettings;
+        return this.cachedFrontendSettings!;
```

Comment:
Why did you add not (!) here?

Reply:
It's just telling the compiler that this expression will not be null or undefined. I'm not sure that's entirely true, but we are assuming that fetchFrontendSettings always returns a value. In getUIConfig this line https://github.com/Unleash/unleash/blob/2553bf18e55c0730e997c33529fd815301b0fdd8/src/lib/routes/admin-api/config.ts#L169 would fail.

**Another option** is to change the signature of the method to be 
```typescript
async fetchFrontendSettings(): Promise<FrontendSettings | undefined>
```
and handle this case on the clients.

**Third option**: just leave the compiler error alone (note it's actually an error, but we're ignoring null-checks. This is one of those cases). I think this is a viable option because this line is unrelated to this PR change (just something I found while working on this file).

Reply:
but returning `!undefined` will be "true" which in my book is not the same as `FrontendSettings`? Would it not be better to deal with the possible null/undefined and return a default value instead?

Reply:
Another option: Not try to catch the error. We would have to control what the memoziee function should do then.

Reply:
Like Gastón said, this is a [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) in TS, which basically is "just telling the compiler that this expression will not be null or undefined". It's not the same as `!this.cachedFrontendSettings`.

I think it's good scouting since we're keeping the behavior we had previously. The difference is that we're "fixing" the TS error by assuring that we're passing a non-null value.

If we want to fix this for good I would look into why `cachedFrontendSettings` is possibly null/undefined and fix the types there, or possibly add a logic branch for when it is, like `if (!this.cachedFrontendSettings) throw new Error('No frontend settings found.')` before.

As it stands I'm happy with merging this as is, since, like mentioned before, "we're keeping the behavior we had previously."

Reply:
@ivarconr `!` in front of the expression is a javascript operater and will negate the value of the expression. `!` after the expression is way of telling the typescript compiler that the value will always exist: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Reply:
Leaving this change out just to move this PR forward